c_src/extractor.cc

```diff
@@ -338,6 +338,18 @@ DataType::Type Extractor::cTypeOf(ErlNifEnv* env, ERL_NIF_TERM tuple, bool throw
  * Given the start of a key data binary, return the start and size of the
  * contents portion of an encoded riak object
  */
+bool Extractor::riakObjectContentsCanBeParsed(const char* data, size_t size)
+{
+    const char* ptr = data;
+    ptr++;
+    unsigned char vers     = (*ptr++);
```

Comment:
There is a magic number at offset 0: https://github.com/basho/riak_kv/blob/develop/src/riak_object.erl#L79. Perhaps you should be checking it as well?